$NetBSD: patch-CMakeLists.txt,v 1.3 2011/10/28 17:34:07 schnoebe Exp $

#
# change man path to be configurable to PKGSRC standards.
# silence a warning from cmake
# remove -export-dynamic which fails on SunOS
#
--- CMakeLists.txt.orig	2011-06-11 13:17:44.000000000 +0000
+++ CMakeLists.txt
@@ -3,6 +3,7 @@ project(spectrum)
 cmake_minimum_required(VERSION 2.6.0 FATAL_ERROR)
 if(COMMAND cmake_policy)
 	cmake_policy(SET CMP0003 NEW)
+	cmake_policy(SET CMP0017 OLD)
 endif(COMMAND cmake_policy)
  
 file(WRITE src/transport_config.h "\n")
@@ -80,9 +81,9 @@ FILE(GLOB spectrum_SRCS src/*.cpp src/pr
 add_executable(spectrum ${spectrum_SRCS} ${lrelease_outputs})
 
 if(DEFINED WITH_STATIC_GLOOX)
-	target_link_libraries(spectrum ${WITH_STATIC_GLOOX} -lgnutls ${PURPLE_LIBRARY} ${GLIB2_LIBRARIES} ${LIBPOCO_LIBRARIES} ${EVENT_LIBRARIES} ${Magick_LIBRARY} -export-dynamic)
+	target_link_libraries(spectrum ${WITH_STATIC_GLOOX} -lgnutls ${PURPLE_LIBRARY} ${GLIB2_LIBRARIES} ${LIBPOCO_LIBRARIES} ${EVENT_LIBRARIES} ${Magick_LIBRARY})
 else(DEFINED WITH_STATIC_GLOOX)
-	target_link_libraries(spectrum ${GLOOX_LIBRARIES} ${PURPLE_LIBRARY} ${GLIB2_LIBRARIES} ${LIBPOCO_LIBRARIES} ${EVENT_LIBRARIES} ${Magick_LIBRARY} -export-dynamic)
+	target_link_libraries(spectrum ${GLOOX_LIBRARIES} ${PURPLE_LIBRARY} ${GLIB2_LIBRARIES} ${LIBPOCO_LIBRARIES} ${EVENT_LIBRARIES} ${Magick_LIBRARY})
 endif(DEFINED WITH_STATIC_GLOOX)
 
 message(STATUS "Transport will be installed into: " ${CMAKE_INSTALL_PREFIX})
@@ -176,11 +177,11 @@ INSTALL(FILES
 
 INSTALL(FILES
     ${CMAKE_CURRENT_SOURCE_DIR}/man/spectrumctl.8
-	DESTINATION share/man/man8
+	DESTINATION @PKGMANDIR@/man8
     )
 INSTALL(FILES
     ${CMAKE_CURRENT_SOURCE_DIR}/man/spectrum.cfg.5
-	DESTINATION share/man/man5
+	DESTINATION @PKGMANDIR@/man5
     )
 
 endif (NOT WITHOUT_SPECTRUMCTL)
